### ISSUE TITLE
Valid double ampersand fails to compile

### ISSUE THREAD

**Issue by tarikhamilton** [leafo/scssphp#556]
In SCSS, you can't have two ampersands next to each other, but if you put it in the interpolation brackets `#{&}`, you can have them side-by-side. My SCSS compiles via Node Sass just fine, but when using scssphp, I get a compile error.

```
.test {
    &#{&}--alt {
        color: red;
    }
}
```

The work around for now is to just use a variable, but it isn't ideal and I have to adjust my SCSS to work with scssphp.

```
$test: 'test';

#{$test} {
    &#{$test}--alt {
        color: red;
    }
}
```

Quickly writing this at work, but let me know if there is anything I can do to help. Thanks for the awesome library!

**Comment by robocoder** [comment]
This looks like a sass 3.4 feature with parent selector.

sass3.3.14 outputs:
```
Invalid CSS after "    &#{": expected expression (e.g. 1px, bold), was "&}--alt {"
```

**Comment by niaccurshi** [comment]
This should be resolved as part of #593

**Comment by Cerdic** [comment]
indeed, this issue is resolved
